Sources/InteroperabilityTests/TestService.swift

```diff
@@ -142,6 +154,18 @@ public struct TestService: Grpc_Testing_TestService.ServiceProtocol {
     var aggregatedPayloadSize = 0
 
     for try await message in request.messages {
+      // We can't validate messages at the wire-encoding layer (i.e. where the compression byte is
+      // set), so we have to check via the encoding header. Note that it is possible for the header
+      // to be set and for the message to not be compressed.
+      if message.expectCompressed.value,
+        !request.metadata.contains(where: { $0.key == "grpc-encoding" })
```

Comment:
nit: can we pull this scan out of the loop so that we don't do it for each message?